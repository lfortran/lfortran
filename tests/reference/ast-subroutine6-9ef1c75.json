{
    "basename": "ast-subroutine6-9ef1c75",
    "cmd": "lfortran --indent --show-ast --no-color {infile} -o {outfile}",
    "infile": "tests/subroutine6.f90",
    "infile_hash": "1987bc648e7f4c6205190248a296d000e0f33cb57a996b6c2a6c4804",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "ast-subroutine6-9ef1c75.stdout",
    "stdout_hash": "30ada3b5058dbad3250151e8fdc560d7f9e4fb0c39532468f91858b5",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}