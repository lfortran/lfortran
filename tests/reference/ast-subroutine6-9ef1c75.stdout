(TranslationUnit 
    [(Subroutine 
        triad 
        [(a)
        (b)
        (scalar)
        (c)] 
        [] 
        () 
        () 
        [] 
        [] 
        [] 
        [(Declaration 
            (AttrType 
                TypeReal 
                [] 
                () 
                None
            ) 
            [(AttrIntent 
                In
            )] 
            [(a 
            [(() 
            () 
            DimensionExpr)] 
            [] 
            () 
            None 
            ())
            (b 
            [(() 
            () 
            DimensionExpr)] 
            [] 
            () 
            None 
            ())
            (scalar 
            [] 
            [] 
            () 
            None 
            ())] 
            ()
        )
        (Declaration 
            (AttrType 
                TypeReal 
                [] 
                () 
                None
            ) 
            [(AttrIntent 
                Out
            )] 
            [(c 
            [(() 
            () 
            DimensionExpr)] 
            [] 
            () 
            None 
            ())] 
            ()
        )
        (Declaration 
            (AttrType 
                TypeInteger 
                [] 
                () 
                None
            ) 
            [] 
            [(N 
            [] 
            [] 
            () 
            None 
            ())
            (N2 
            [] 
            [] 
            () 
            None 
            ())
            (i 
            [] 
            [] 
            () 
            None 
            ())
            (j 
            [] 
            [] 
            () 
            None 
            ())] 
            ()
        )] 
        [(= 
            0 
            N 
            (FuncCallOrArray 
                size 
                [] 
                [(() 
                a 
                ())] 
                [] 
                []
            ) 
            ()
        )
        (= 
            0 
            N2 
            (FuncCallOrArray 
                size 
                [] 
                [(() 
                b 
                ())] 
                [] 
                []
            ) 
            ()
        )
        (DoConcurrentLoop 
            0 
            () 
            [(ConcurrentControl 
                i 
                1 
                N 
                ()
            )] 
            () 
            [] 
            [(= 
                0 
                (FuncCallOrArray 
                    c 
                    [] 
                    [(() 
                    i 
                    ())] 
                    [] 
                    []
                ) 
                (+ (FuncCallOrArray 
                    a 
                    [] 
                    [(() 
                    i 
                    ())] 
                    [] 
                    []
                ) (* scalar (FuncCallOrArray 
                    b 
                    [] 
                    [(() 
                    i 
                    ())] 
                    [] 
                    []
                ))) 
                ()
            )] 
            () 
            (TriviaNode 
                [] 
                [(EndOfLine)
                (EndOfLine)]
            )
        )
        (DoConcurrentLoop 
            0 
            () 
            [(ConcurrentControl 
                j 
                1 
                N2 
                ()
            )] 
            () 
            [] 
            [(= 
                0 
                (FuncCallOrArray 
                    c 
                    [] 
                    [(() 
                    j 
                    ())] 
                    [] 
                    []
                ) 
                (+ (FuncCallOrArray 
                    b 
                    [] 
                    [(() 
                    j 
                    ())] 
                    [] 
                    []
                ) scalar) 
                ()
            )] 
            () 
            ()
        )] 
        []
    )]
)
